contents/docs/libraries/js/index.mdx

```diff
@@ -444,30 +444,44 @@ It has two optional parameters:
 
--   `persistence: "cookie"` (default). Everything is stored in a cookie.
--   `persistence: "localStorage+cookie"`. User's distinct ID is stored in a cookie and everything else is stored in the browser's <a href="https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage" target="_blank">localStorage</a>.
--   `persistence: "localStorage"`. Everything is stored in localStorage.
--   `persistence: "sessionStorage"`. Everything is stored in sessionStorage.
--   `persistence: "memory"`. Stores in page memory, which means data is only persisted for the duration of the page view.
+- `persistence: "cookie"` (default): Stores all data in a cookie.
 
-As a note, due to the size limitation of cookies you may run into `431 Request Header Fields Too Large` errors (e.g. if you have a lot of feature flags). In that case, use `localStorage+cookie`.
+- `persistence: "localStorage+cookie"`: Stores user's distinct ID in a cookie and everything else in the browser's <a href="https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage" target="_blank">`localStorage`</a>.
 
-> **Note: ** Please be aware that localStorage and sessionStorage can't be used across subdomains. If you have multiple sites on the same domain, you may want to consider the <code>cookie</code> option or make sure to set all super properties across each subdomain.
+- `persistence: "localStorage"`: Stores everything in `localStorage`.
 
-If you don't want PostHog to store anything on the user's browser (e.g. if you want to rely on your own identification mechanism only, or want completely anonymous users), you can set `disable_persistence: true` in PostHog's config.
-**Warning:** Remember to call `posthog.identify` every time your app loads or every page refresh will be treated as a different user.
+- `persistence: "sessionStorage"`: Stores everything in `sessionStorage`.
+
+- `persistence: "memory"`: Stores everything in page memory, which means data is only persisted for the duration of the page view.
+
+To change `persistence` values without reinitializing PostHog, you can use the `posthog.set_config()` method. This enables you to switch from memory to cookies to better comply with privacy regulations.
+
+```js-web
+const handleCookieConsent = (consent) => {
+    posthog.set_config({ persistence: consent === 'yes' ? 'localStorage+cookie' : 'memory' });
+    localStorage.setItem('cookie_consent', consent);
+};
+```
+
+### Persistence caveats
+
+- Be aware that `localStorage` and `sessionStorage` can't be used across subdomains. If you have multiple sites on the same domain, you may want to consider a `cookie` option or make sure to set all super properties across each subdomain.
+
+- Due to the size limitation of cookies you may run into `431 Request Header Fields Too Large` errors (e.g. if you have a lot of feature flags). In that case, use `localStorage+cookie`.
```

Comment:
when would I run into this issue?

Reply:
if you have a lot of feature flags (or a lot of cookies), I've run into it